app/src/proxy.ts

```diff
@@ -1,57 +1,125 @@
 import child_process from "node:child_process";
+import fs from "fs";
+import path from "path";
+
 import { JSONObject } from "./utils";
+import { Writable } from "node:stream";
 
 export type ProxyRequest = {
   method: "GET" | "POST" | "DELETE";
   path_query: string;
-  stream: false;
+  stream: boolean;
   body?: string;
   headers: object;
 };
 
-export type ProxyResponse = {
+export type ProxyCommand = {
+  command: string;
+  options: string[];
+  proxyOrigin: string;
+  timeoutMs: number;
```

Comment:
Nit:  This might be a good opportunity to consider documenting and enforcing units via [nominal typing](https://www.learningtypescript.com/articles/branded-types#branded-numbers), e.g.:

```typescript
type ms = number & { readonly __unit: 'ms' };

// ...

export type ProxyCommand = {...,
  timeout: ms,
  ...
};
```

What do you think, @vickiniu and @legoktm?  Might be overkill, but I thought I'd mention it early, while we're still accumulating coding standards and style in TypeScript.  :-)